pkg/application/service/factory/service_factory.go

```diff
@@ -88,6 +88,12 @@ func (s *ServiceFactory) WithVerificationServiceOption(opt verificationservice.V
 	s.verificationServiceOptions = append(s.verificationServiceOptions, opt)
 }
 
+func (s *ServiceFactory) WithSignupService(signupService service.SignupService) {
+	s.signupServiceFunc = func(_ ...signupservice.SignupServiceOption) service.SignupService {
+		return signupService
+	}
+}
```

Comment:
This is a temp hack to inject specific fake version of signupService without any extra abstraction and additional mocking.